app/src/main/java/com/rud/movieapp/MovieApplication.kt

```diff
@@ -0,0 +1,24 @@
+package com.rud.movieapp
+
+import android.app.Activity
+import android.app.Application
+import com.rud.movieapp.di.AppInjector
+import dagger.android.AndroidInjector
+import dagger.android.DispatchingAndroidInjector
+import dagger.android.HasActivityInjector
+import timber.log.Timber
+import javax.inject.Inject
+
+class MovieApplication : Application(), HasActivityInjector {
+
+    @Inject
+    lateinit var dispatchingActivityInjector: DispatchingAndroidInjector<Activity>
+
+    override fun activityInjector(): AndroidInjector<Activity> = dispatchingActivityInjector
```

Comment:
you can exteds from DaggerApplication. In this case you will not need define AndroidInjector and implement HasActivityInjector